cmake/HPX_DetectCppDialect.cmake

```diff
@@ -89,24 +89,20 @@ macro(hpx_detect_cpp_dialect)
 
     # enable enforcing a particular C++ mode
     if(HPX_WITH_CXX17)
-      set(CXX_FLAG -std:c++latest)
-      hpx_add_config_cond_define(_HAS_AUTO_PTR_ETC 1)
+      set(CXX_FLAG -std:c++17)
       hpx_info("C++ mode enforced: C++17")
     elseif(HPX_WITH_CXX1Z)
       set(CXX_FLAG -std:c++latest)
```

Comment:
C++1z essentially is C++17, right?

Reply:
Well, not quite - doesn't really change anything, however.

Reply:
my point is, when the next version of MSVC is released, `-std:c++latest` might refer to C++2a, right?

Reply:
Who knows.